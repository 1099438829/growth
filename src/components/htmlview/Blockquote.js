import React, { Component, PropTypes } from 'react';
import { View } from 'react-native';
import AppStyle from '../../theme/styles';

class Blockquote extends Component {
  static componentName = 'Blockquote';

  static propTypes = {
    blockquoteStyle: View.propTypes.style,
    component: PropTypes.node,
  };

  static defaultProps = {
    blockquoteStyle: AppStyle.blockquote,
    component: null,
  };
  render() {
    return <View style={this.props.blockquoteStyle}>{this.props.component}</View>;
  }
}
export default Blockquote;
