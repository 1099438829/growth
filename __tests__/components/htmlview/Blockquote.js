import { Text } from 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import Blockquote from '../../../src/components/htmlview/Blockquote';

it('renders correctly', () => {
  // eslint-disable-next-line no-unused-vars
  const blockquote = renderer.create(
    <Blockquote component={<Text>2333</Text>} />,
  );
  const blockquoteJson = blockquote.toJSON();
  expect(blockquoteJson).toMatchSnapshot();
});
