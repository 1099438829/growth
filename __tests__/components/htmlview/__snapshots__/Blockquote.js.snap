// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderLeftColor": "#dadada",
      "borderLeftWidth": 3,
      "flex": 1,
      "marginTop": 15,
      "paddingLeft": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    2333
  </Text>
</View>
`;
